lib/auth/trustedcluster.go

```diff
@@ -67,8 +68,15 @@ func (a *Server) UpsertTrustedCluster(ctx context.Context, trustedCluster types.
 	}
 
 	// change state
+
+	if exists && !cmp.Equal(existingCluster.GetRoleMap(), trustedCluster.GetRoleMap()) {
```

Comment:
nit: we can remove `exists` with `existingCluster != nil`; the check for `exists` should also probably only explicitly ignore a `NotFound` rather than blanket swallow all errors.